### ISSUE TITLE
[BUG] - [P2P] systemd socket activation does not work correctly.

### ISSUE THREAD

**Issue by jbgi** [input-output-hk/cardano-node#2721]
When using systemd socket activation, the node fails to connect to other nodes:
```
TracePromoteColdFailed 20 0 18.133.216.203:3001 5s Network.Socket.bind: resource busy (Address already in use)
```
Disabled for now.

**Comment by karknu** [comment]
Can you share you socket activation config file?

**Comment by jbgi** [comment]
systemd socket unit file:
```
[Unit]
Description=Socket of the cardano-node service.
PartOf=cardano-node.service

[Socket]
FreeBind=yes
ListenStream=172.31.39.95:3001
ListenStream=/run/cardano-node/node.socket
ReusePort=yes
RuntimeDirectory=cardano-node
SocketGroup=cardano-node
SocketMode=0660
SocketUser=cardano-node
```

**Comment by karknu** [comment]
I'm unable to reproduce the issue on `4513d36a8ce37a9f707fa2038cf8861611b39cef` in p2p-master. Socket activation appears to be working.

**Comment by jbgi** [comment]
Still having the issue with latest p2p-master (`2715e45eb7f2914e8010129c67d1bf0081123110`). Seeing lots of 
`TracePromoteColdFailed 40 0 3.67.149.107:3001 159.019085325783s Network.Socket.bind: resource busy (Address already in use)` and no block exchanged when socket activation is used on all nodes.

**Comment by karknu** [comment]
I'm still unable to reproduce this.
Are you sure you don't have another process (cardano-node instance) using port 3001?

**Comment by coot** [comment]
Socket activation seems to work for me for p2p nodes, please re-open if you still have a problem with it.  I've been using the following `*.socket` file:

```systemd
[Socket]
ListenStream=0.0.0.0:7000
NoDelay=true
ReusePort=true
FreeBind=true

[Socket]
ListenStream=[::]:7000
NoDelay=true
ReusePort=true
BindIPv6Only=true
FreeBind=true

[Socket]
ListenStream=/data/cardano/node/cardano-node.socket
RemoveOnStop=true
```

We still need to update `cardano-node` which will fix input-output-hk/ouroboros-network#3977.

**Comment by waalge** [comment]
I just hit this with any environment bar mainnet 

```nix
# ...
services.cardano-node = {
  hostAddr = "0.0.0.0";
}
# ...
```
Seems to get around it.
